src/libraries/System.Net.NetworkInformation/tests/FunctionalTests/NetworkAddressChangedTests.cs

```diff
@@ -23,5 +26,56 @@ public void NetworkAddressChanged_JustRemove_Success()
         {
             NetworkChange.NetworkAddressChanged -= _addressHandler;
         }
+
+        [PlatformSpecific(TestPlatforms.Linux)]
+        [Fact]
+        public async void NetworkAddressChanged_AddRemoveMultipleTimes_CheckForLeakingThreads()
+        {
+            static async Task<int> GetNumberOfNetworkAddressChangeThreadsAsync()
+            {
+                int pid = Process.GetCurrentProcess().Id;
+                ProcessStartInfo psi = new ProcessStartInfo("ps", $"-T -p {pid}");
```

Comment:
Unfortunately calling `ps` fails on some of our docker images:
```
System.Net.NetworkInformation.Tests.NetworkChangedTests.NetworkAddressChanged_AddRemoveMultipleTimes_CheckForLeakingThreads [FAIL]
      System.ComponentModel.Win32Exception : An error occurred trying to start process 'ps' with working directory '/root/helix/work/workitem/e'. No such file or directory
```
I need to follow up with this with our infra, whether to condition the test or fix the images...

Reply:
Yes, I noticed few tests were failing because "ps" was missing. I don't know if it is good practice to call external programs from tests. Unfortunately I don't think such test can be done with C# managed code.

Reply:
We do have some precedent for calling external process from a test and I don't think there's another (managed) way to count the threads by their names.
I found an example of a check that tests a similar thing:
https://github.com/dotnet/runtime/blob/5a12420546d7f0d963bcccf082400c1b057e38ef/src/libraries/System.Net.Security/tests/FunctionalTests/ServerAllowNoEncryptionTest.cs#L74-L74
And the implementation of the check leads to this:
https://github.com/dotnet/runtime/blob/5a9e584d684d45ce67b6e9c9cf27114db20efd1b/src/libraries/System.Net.Security/tests/FunctionalTests/TestConfiguration.cs#L36

Basically trying to call the required CLI command and if it fails disabling the tests that depends on it.

Do you think you could add something like this for this test?

Reply:
For reference, issue to add `ps` into the docker image: https://github.com/dotnet/dotnet-buildtools-prereqs-docker/issues/565

Reply:
Hi,

Thank you for this. I think this should now be implemented with https://github.com/dotnet/runtime/pull/63963/commits/d89300329c992c981bd984fa35e8915767f57673 and https://github.com/dotnet/runtime/pull/63963/commits/e94ab24bb170cacd80f518d23049bd7999b75bc2

Also, [OuterLoop()] should be probably enabled for this test as you suggested and remove TODO.